packages/core-bridge/ts/errors.ts

```diff
@@ -27,6 +28,35 @@ export class UnexpectedError extends Error {
   }
 }
 
+export interface NativeServiceError {
+  name: 'ServiceError';
+  message: string;
+  code: number;
+  details: string;
+  metadata: Record<string, Buffer | string>;
+  stack?: string;
+}
+
+@SymbolBasedInstanceOfError('ServiceError')
+export class ServiceError extends Error implements grpc.ServiceError {
```

Comment:
I would ideally merge TransportError and ServiceError, they're different representations of the tonic error.

You can remove this new error type and add the new properties to `TransportError`.

Reply:
I was too lazy to do this as part of my PR but with your `from` and `into` helpers, it should be fairly straightforward.